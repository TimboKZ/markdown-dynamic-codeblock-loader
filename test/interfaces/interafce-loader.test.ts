import { expect, test } from '@jest/globals';
import path from 'path';

import compiler from '../compiler';

test('Inserts name and outputs JavaScript', async () => {
    const fixturePath = path.resolve(__dirname, 'raw-input.md');
    const stats = await compiler(fixturePath);

    const output = stats.toJson().modules![0].source;

    let markdownOutput = '';
    if (output) {
        markdownOutput = output.replace(/^export default/, '').replace(/;$/, '');
    }
    markdownOutput = JSON.parse(markdownOutput);

    expect(markdownOutput).toMatchSnapshot();
});
