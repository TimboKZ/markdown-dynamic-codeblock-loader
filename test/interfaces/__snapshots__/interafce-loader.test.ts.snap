// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inserts name and outputs JavaScript 1`] = `"export default \\"# Interface tests raw input file\\\\n\\\\nThis file is used to test that the loader correctly replaces the code blocks below\\\\nwith the source snippets referenced in code block config.\\\\n\\\\n\`\`\`ts { \\\\\\"file\\\\\\", \\\\\\"../typescript-source/typedef.ts\\\\\\", \\\\\\"symbol\\\\\\": \\\\\\"MyData\\\\\\" }\\\\n\`\`\`\\\\n\\\\n\\""`;
